### PR TITLE
GODRIVER-1999 allow directConnection=false in loadBalanced=true URIs

### PR DESCRIPTION
**Pull request by iwysiu**

### CODE REVIEW

**Review by matthewdale — APPROVED** [review]
Looks good 👍

File: data/initial-dns-seedlist-discovery/load-balanced/loadBalanced-directConnection.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# The TXT record for test20.test.build.10gen.cc contains loadBalanced=true.
```
[the remark this answers is not in the document]
Comment (iwysiu) [reply data/initial-dns-seedlist-discovery/load-balanced/loadBalanced-directConnection.yml:1]: All of the test files in specifications are written in yaml and then compiled to json, and these folders are just copied over from their respective folders in the spec directory.

File: x/mongo/driver/connstring/connstring.go [hunk 2]
```diff
@@ -340,6 +340,9 @@ func (p *parser) validate() error {
 		if p.Scheme == SchemeMongoDBSRV {
 			return errors.New("a direct connection cannot be made if an SRV URI is used")
 		}
+		if p.LoadBalancedSet && p.LoadBalanced {
```
Comment (benjirewis) [inline x/mongo/driver/connstring/connstring.go:343]: Makes more sense to have this check here since its an "invalid use of direct connections".

**Review by benjirewis — APPROVED** [review]
LGTM. I've also wondered about our inclusion of YAML files for tests, @matthewdale. They do help for quick readability, but most other drivers only keep the JSON around 🤷

**Pull request merged**